File: NodeJS/login_with_amazon.js
var querystring = require('querystring')
var http = require('https')

function Login(client_id, sandbox) {
  this.client_id = client_id
  this.sandbox = sandbox
}

Login.prototype = {
	constructor: Login
	, getLoginProfile: function(accessToken, callback) {
		var decodedAccessToken = decodeURIComponent(accessToken)
		var encodeAccessToken = encodeURIComponent(decodedAccessToken)

		if (this.sandbox == true) {
		  var sandbox_str = "api.sandbox"
		} else {
		  var sandbox_str = "api"
		}

		var client_id = this.client_id

		http.get("https://" + sandbox_str + ".amazon.com/auth/o2/tokeninfo?access_token=" + encodeAccessToken, function(res) {
			res.on('data', function (chunk) {
				if (JSON.parse(chunk)['aud'] != client_id) {
					throw new Error('Invalid Access Token')
				}

				var get_options = {
				  host: sandbox_str+".amazon.com",
				  path: "/user/profile",
				  method: 'GET',
				  headers: {
				    'Authorization': "bearer " + decodedAccessToken
				  }
				}

				var get_req = http.request(get_options, function(res) {
					res.setEncoding('utf8')
				  res.on('data', function (chunk) {
				    callback(JSON.parse(chunk))
				  }).on('error', function(e) {
						callback(e.message)
					})
				})

				get_req.end()

			})
		}).on('error', function(e) {
		  callback(e.message)
		})

    process.on('uncaughtException', function (err) {
     callback(err)
    })
	}
}

module.exports = Login;

File: Ruby/login_with_amazon.rb
require 'amazon_pay'

# Your client id is located in your Seller
# Central account. Be sure to store your keys
# in a secure location. 
client_id = 'Your Client Id'

login = AmazonPay::Login.new(
  client_id,
  region: :na,  # Default: :na
  sandbox: true # Default: false
)

# The access token is available in the return URL
# parameters after a user has logged in.
access_token = 'User Access Token'

# Make the 'get_login_profile' api call.
profile = login.get_login_profile(access_token)

name = profile['name']
email = profile['email']
user_id = profile['user_id']
